src/components/Menus/MenuItem.js

```diff
@@ -0,0 +1,207 @@
+    display: 'table-cell',
+    verticalAlign: 'middle',
+    paddingLeft: '16px'
+  },
+  label: {
+    padding: '0 16px',
+    whiteSpace: 'nowrap'
+  },
+  leftIconStyles: {
+    fontSize: '22px'
+  }
+}
+
+@Radium
+class MenuItem extends React.Component {
+  static propTypes = {
+    /** Completely override the MenuItem rendering and create a custom MenuItem */
+    children        : PropTypes.node,
+    /** Disable the MenuItem */
+    disabled        : PropTypes.bool,
+    /** Focus the MenuItem */
+    focus           : PropTypes.bool,
+    /** Index of MenuItem within Menu. Used for currentIndex */
+    index           : PropTypes.number,
+    /** Text for the menu item */
+    label           : PropTypes.string.isRequired,
+    /** Override styles of label */
+    labelStyles     : PropTypes.shape({}),
+    /** Icon name or Icon component displayed left of label*/
+    leftIcon        : PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.element
+    ]),
+    /** Override styles for leftIcon */
+    leftIconStyles  : PropTypes.shape({}),
+    /** Callback function fired when the menu item is selected. Sends option object as {value:, label:} and index */
+    onClick         : PropTypes.func,
+    /** Callback function fired when the menu item is focused. */
+    onFocus         : PropTypes.func,
+    /** Used by menu to keep track of current focus index. */
+    onMenuItemFocus : PropTypes.func,
+    /** Role HTML attribute */
+    role            : PropTypes.string,
+    /** Customize style of MenuItem */
+    style           : PropTypes.shape({}),
+    /** Override tabIndex property*/
+    tabIndex        : PropTypes.number,
+    /** Whether or not to use use the tabIndex HTML attribute */
+    useTabIndex     : PropTypes.bool,
+    /** Underlying value. Also, passed into onClick function */
+    value           : PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+      PropTypes.bool
+    ]).isRequired,
+  }
+
+  static defaultProps = {
+    disabled: false,
+    role: 'menuitem',
+    useTabIndex: true,
+    focus: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { focus } = this.props
+    if (!focus && nextProps.focus) {
+      setTimeout( () => {
+        this.menuItem.focus()
+      }, 0)
+    }
+  }
+
+  _handleClick = () => {
+    const {
+      disabled,
+      index,
+      onClick,
+      value,
+      label
+    } = this.props
+
+    const option = {value: value, label: label}
+
+    !disabled && onClick && onClick(option, index)
+  }
+
+  _handleKeyDown = (event) => {
+    switch(event.key) {
+      case 'Enter':
+        event.preventDefault()
+        this._handleClick()
+    }
+  }
+
+  _handleFocus = (event) => {
+    const { onFocus, onMenuItemFocus, index } = this.props
+    onMenuItemFocus && onMenuItemFocus(index)
+    onFocus && onFocus(index, event)
+  }
+
+  renderLeftIcon() {
+    const { leftIcon, leftIconStyles } = this.props
+    let iconComponent = null
```

Comment:
I'm a big fan of early returns to simplify logic/avoid mutating variables:
```
if (!leftIcon) { return }
const iconComponent = React.isValidElement(leftIcon) ? leftIcon : <Icon name={leftIcon} style={[styles.leftIconStyles, leftIconStyles]} />
```

Reply:
Actually I think we should reverse the conditional so that we get better error messages if `leftIcon` is not a valid component:
```
const iconComponent = typeof leftIcon === 'String'
  ? <Icon name={leftIcon} style={[styles.leftIconStyles, leftIconStyles]} />
  : leftIcon
```